langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -1060,6 +1060,73 @@ function testLangLibFuncWithNamedArgExpr() {
     }
 }
 
+function testEmptyRegexpFind() {
+    // find
+    regexp:Span? resA1 = regexp:find(re = re ``, str = "HelloWorld");
+    assertTrue(resA1 is null);
+    regexp:Span? resA2 = regexp:find(re = re `World`, str = "");
+    assertTrue(resA2 is null);
+    regexp:Span? resA3 = regexp:find(re = re ``, str = "");
+    assertTrue(resA3 is null);
+    regexp:Span? resA4 = regexp:find(re = re `${""}`, str = "");
+    assertTrue(resA4 is null);
+    regexp:Span? resA5 = regexp:find(re = re `${""}`, str = "HelloWorld");
+    assertTrue(resA5 is null);
+    string regexStrA = "";
+    regexp:Span? resA6 = regexp:find(re = re `${regexStrA}`, str = "HelloWorld");
+    assertTrue(resA6 is null);
+    regexp:Span? resA7 = regexp:find(re = re `${regexStrA}`, str = "");
+    assertTrue(resA7 is null);
+    regexp:Span? resA8 = regexp:find(re = re `(.*)`, str = "");
+    assertTrue(resA8 is regexp:Span);
+
+    // find all
+    regexp:Span[] resB1 = regexp:findAll(re ``, "There once was a king who liked to sing");
+    assertEquality(0, resB1.length());
+    regexp:Span[] resB2 = regexp:findAll(re `(\w+ing)`, "");
+    assertEquality(0, resB2.length());
+    regexp:Span[] resB3 = regexp:findAll(re ``, "");
+    assertEquality(0, resB3.length());
+    regexp:Span[] resB4 = regexp:findAll(re `${""}`, "");
+    assertEquality(0, resB4.length());
+    regexp:Span[] resB5 = regexp:findAll(re `${""}`, "There once was a king who liked to sing");
+    assertEquality(0, resB5.length());
+
+   //  // find groups
+    regexp:Groups? resC1 = regexp:findGroups(re ``, "Butter was bought by Betty but the butter was bitter");
+    assertTrue(resC1 is null);
```

Comment:
Don't we need to use `resC1 is ()` instead of `resC1 is null`

Reply:
Will be addressed aligning with the new approach mentioned at: https://github.com/ballerina-platform/ballerina-lang/pull/39444#discussion_r1124231161.